docs/architecture.md

```diff
@@ -36,13 +36,15 @@ This is the mechanism that traffic from live users enters your cluster and is re
 Argo Rollouts is very flexible on networking options. First of all you can have different services during a Rollout, that go only to the new version, only to the old version or both.
 Specifically for Canary deployments, Argo Rollouts supports several [service mesh and ingress solutions](../features/traffic-management/) for splitting traffic with specific percentages instead of simple balancing based on pod counts.
 
-## Analysis and AnalysisRun
+## AnalysisTemplate and AnalysisRun
 
-Analysis is a custom Kubernetes resource that connects a Rollout to your metrics provider and defines specific thresholds for certain metrics that will decide if a rollout is successful or not. For each Analysis you can define one or more metric queries along with their expected results. A Rollout will progress on its own if metric queries are good, rollback automatically if metrics show failure and pause the rollout if metrics cannot provide a success/failure answer.
+Analysis is the capability to connect a Rollout to your metrics provider and define specific thresholds for certain metrics that will decide if a rollout is successful or not. For each analysis you can define one or more metric queries along with their expected results. A Rollout will progress on its own if metric queries are good, rollback automatically if metrics show failure and pause the rollout if metrics cannot provide a success/failure answer.
```

Comment:
- How about ".... a rollout is successful or not" -> ".... an update is successful or not"?

Reply:
@huikang done!